src/inc/internal/Crypto.hpp

```diff
@@ -6,12 +6,45 @@
 
 #include <vector>
 
+#ifndef SHA256_DIGEST_LENGTH

```

Comment:
I am assuming that this can be configurable in the future as a cmake flag in the future?

Reply:
It is just a precaution to prevent redefinition, given the header is included in different source files in different builds. For example, the same MACRO is defined in openssl's sha.h but not in Win32. With this #ifndef, the same MACRO can be used in both source files.